notify/templates.go

```diff
@@ -131,6 +133,26 @@ func (am *GrafanaAlertmanager) TestTemplate(ctx context.Context, c TestTemplates
 	return &results, nil
 }
 
+func (am *GrafanaAlertmanager) GetTemplate() (*template.Template, error) {
+	seen := make(map[string]struct{})
+	tmpls := make([]string, 0, len(am.templates))
+	for _, tc := range am.templates {
+		if _, ok := seen[tc.Name]; ok {
+			level.Warn(am.logger).Log("msg", "template with same name is defined multiple times, skipping...", "template_name", tc.Name)
+			continue
+		}
+		tmpls = append(tmpls, tc.Template)
+		seen[tc.Name] = struct{}{}
+	}
```

Comment:
I can't see where we did this before, does it matter if we have templates with duplicate names? Wondering if this will break for people that do have duplicates.